src/Microsoft.Sbom.Extensions/ISbomParser.cs

```diff
@@ -58,7 +58,7 @@ public interface ISbomParser
     /// </summary>
     /// <param name="stream"></param>
     /// <returns></returns>
-    SBOMMetadata GetMetadata();
+    Spdx22Metadata GetMetadata();
```

Comment:
Yeah, maybe Spdx22Metadata should extend from an `ISBOMMetadata` or something?

Reply:
Yeah that was my bigges pet peeve about this. This should technically be agnostic to the format... however, the metadata included is very much tied to spdx22, so couldn't come up with a better name.

In the future, if spdx3 comes around, we should create a base class with the common properties, and spdx22metadata and spdx30metadata classes with additional properties. The API then should return the base class. 

Its not SBOMMetadata, since that has a different meaning in our world, its used for generation.

Reply:
Hmm, maybe we could call this `GetSpdx22Metadata` then to future-proof ourselves?

Reply:
I would have made this method generic to return T. The entire interface would be ISbomParser<T> where T will be Spdx22Metadata  for this specific implementation. We will need to add ISbomMetadata interface or something to restrict T.